CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
There is still a larger maintenance burden of course, since external dependencies can still introduce warnings in other headers. I'm now wondering again if including everything external with `SYSTEM` isn't actually the best option.

An additional complication is that I don't get any warnings from external dependencies locally, even without the `CMakeLists.txt` modifications and a clean configure+build. I can go by the log from Jenkins of course, but it does illustrate the problem with manually disabling warnings from specific includes.